test/lib/ufe/testPointInstances.py

```diff
@@ -0,0 +1,113 @@
+
+    def setUp(self):
+        self.assertTrue(self._pluginsLoaded)
+
+        mayaUtils.openPointInstancesGrid14Scene()
+
+    def testPointInstances(self):
+        # Create a UFE path to a PointInstancer prim with an instanceIndex on
+        # the end. This path uniquely identifies a specific point instance.
+        ufePath = ufe.Path([
+            mayaUtils.createUfePathSegment('|UsdProxy|UsdProxyShape'),
+            usdUtils.createUfePathSegment('/PointInstancerGrid/PointInstancer/7')])
+        ufeItem = ufe.Hierarchy.createItem(ufePath)
+
+        # Verify that we turned the path into a scene item correctly and that
+        # we can access the PointInstancer prim.
+        self.assertEqual(ufeItem.nodeType(), 'PointInstancer')
+
+        ufeAttrs = ufe.Attributes.attributes(ufeItem)
+        self.assertTrue(ufeAttrs.hasAttribute(UsdGeom.Tokens.positions))
+
+        # XXX: Note that UsdSceneItem currently has no wrapping to Python, so
+        # we are not able to verify that we can query the scene item directly
+        # for its instance index. We can exercise the Utils functions though.
+
+        # XXX: Another note, would be nice here to be able to do
+        # ufePath.string() or str(ufePath) to produce this path string instead.
+        ufePathString = ','.join([str(segment) for segment in ufePath.segments])
+
+        prim = mayaUsdUfe.ufePathToPrim(ufePathString)
+        self.assertTrue(prim)
+        self.assertTrue(prim.IsA(UsdGeom.PointInstancer))
+
+        instanceIndex = mayaUsdUfe.ufePathToInstanceIndex(ufePathString)
+        self.assertEqual(instanceIndex, 7)
+
+        # Tests for the usdPathToUfePathSegment() utility function.
+        ufePathSegmentString = mayaUsdUfe.usdPathToUfePathSegment(prim.GetPrimPath())
+        self.assertEqual(ufePathSegmentString, '/PointInstancerGrid/PointInstancer')
+
+        ufePathSegmentString = mayaUsdUfe.usdPathToUfePathSegment(prim.GetPrimPath(), 7)
+        self.assertEqual(ufePathSegmentString, '/PointInstancerGrid/PointInstancer/7')
+
+        # XXX: Similar to the issue above, Ufe::Path::string() yields a string
+        # *without* the comma separator in between segments, so this string is
+        # not directly usable for creating scene items.
+        ufePrimPathString = mayaUsdUfe.stripInstanceIndexFromUfePath(ufePathString)
+        self.assertEqual(
+            ufePrimPathString,
+            '|world|UsdProxy|UsdProxyShape/PointInstancerGrid/PointInstancer')
+
+        # Create a path to the PointInstancer itself and verify that the
+        # instanceIndex is ALL_INSTANCES.
+        ufePath = ufe.Path([
+            mayaUtils.createUfePathSegment('|UsdProxy|UsdProxyShape'),
+            usdUtils.createUfePathSegment('/PointInstancerGrid/PointInstancer')])
+        ufePathString = ','.join([str(segment) for segment in ufePath.segments])
+
+        instanceIndex = mayaUsdUfe.ufePathToInstanceIndex(ufePathString)
+        self.assertEqual(instanceIndex, -1)
```

Comment:
Would there be extra kosher-ness if the comparison was with ALL_INSTANCES?

Reply:
Agreed.  Maybe this needs a quick code comment?

Reply:
Yup, added a comment here to explain this.

Reply:
Added a comment in 9c5099586fc182b547eb306d728329634e2de3bc.